<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>
  <arg name="standalone" default="true" />

  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg uav_name)_vision_nodelet_manager"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg uav_name)">

    <node name="automatic_start_eagle" pkg="nodelet" type="nodelet" args="$(arg nodelet) mrs_general/AutomaticStartEagle $(arg nodelet_manager)" output="screen">

      <rosparam file="$(find mrs_general)/config/automatic_start_eagle.yaml" />

      <param name="scripts_path" value="$(optenv HOME)" />

      <!-- Subscribers -->
      <remap from="~mavros_state_in" to="mavros/state" />
      <remap from="~tracker_status_in" to="control_manager/tracker_status" />
      <remap from="~mpc_diagnostics_in" to="control_manager/mpc_tracker/diagnostics" />

      <!-- services -->
      <remap from="~takeoff_out" to="uav_manager/takeoff" />
      <remap from="~gofcu_out" to="control_manager/goto_fcu" />

    </node>

  </group>

</launch>
